src/AeroSharp/DataAccess/MapAccess/IMap.cs

```diff
@@ -43,4 +43,13 @@ public interface IMap<TKey, TValue>
     /// <param name="cancellationToken"> A cancellation token to cooperatively cancel the operation. </param>
     /// <returns> A task that represents the asynchronous operation of deleting the map. </returns>
     Task DeleteAsync(CancellationToken cancellationToken);
+
+    /// <summary>
+    ///    Asynchronously Get {key: value} entry where map.value is the ith smallest value where i == rank, starting
+    ///    at zero being the lowest value.
+    /// </summary>
+    /// <param name="rank"> The rank to fetch from the map. </param>
+    /// <param name="cancellationToken"> A cancellation token to cooperatively cancel the operation. </param>
+    /// <returns>A task representing the result of the asynchronous fetch of the rank.</returns>
+    Task<KeyValuePair<TKey, TValue>> GetByRankAsync(int rank, CancellationToken cancellationToken);
```

Comment:
What happens when there is a tie for rank? Should multiple entries be returned here instead (assuming that's how the underlying client handles ties)?

Reply:
So I think I actually cover this somewhat in my integration test.

Reply:
the integration test in `GetByRankAsync_with_the_same_values_returns_the_right_key` was supposed to mimic that, but I wonder if you mean more that both the key and value are the same?

My assumption from the comments of the code is that Aerospike's client is supposed to just return one "item" from GetByRank, not multiple in the case of a tie.  The documentation also mentions that get_by_rank should return just a single item.

https://docs.aerospike.com/server/guide/data-types/cdt-map-ops#get_by_rank

Reply:
Sounds good! Thanks for the docs link.